hawkular-alerts-engine-extensions/hawkular-alerts-aggregated-events-alerter/src/main/java/org/hawkular/alerts/extensions/ExpressionManager.java

```diff
@@ -0,0 +1,181 @@
+import org.hawkular.alerts.api.services.DefinitionsService;
+import org.hawkular.alerts.api.services.ExtensionEventListener;
+import org.hawkular.alerts.api.services.ExtensionsService;
+import org.hawkular.alerts.api.services.PropertiesService;
+import org.jboss.logging.Logger;
+
+/**
+ * Manages the APM expression evaluations and interacts with the Alerts system.
+ *
+ * @author Jay Shaughnessy
+ * @author Lucas Ponce
+ */
+@Startup
+@Singleton
+public class ExpressionManager implements ExtensionEventListener {
+    private final Logger log = Logger.getLogger(ExpressionManager.class);
+
+    private static final String ENGINE_EXTENSIONS = "hawkular-alerts.engine-extensions";
+    private static final String ENGINE_EXTENSIONS_ENV = "ENGINE_EXTENSIONS";
+    private static final String ENGINE_EXTENSIONS_DEFAULT = "true";
+    boolean engineExtensions;
+
+    private static final String TAG_NAME = "HawkularExtension";
+    private static final String TAG_VALUE = "AggregatedEvents";
+    private static final String CONTEXT_PROCESSED = "processed";
+
+    @EJB
+    private PropertiesService properties;
+
+    @EJB
+    private DefinitionsService definitions;
+
+    @EJB
+    private ExtensionsService extensions;
+
+    @EJB
+    private CepEngine cep;
+
+    @Resource
+    private ManagedExecutorService executor;
+
+    @PostConstruct
+    public void init() {
+        engineExtensions = Boolean.parseBoolean(properties.getProperty(ENGINE_EXTENSIONS, ENGINE_EXTENSIONS_ENV,
+                ENGINE_EXTENSIONS_DEFAULT));
+        if (engineExtensions) {
+            log.info("Registering Trigger CREATE/UPDATE/REMOVE listener");
+            definitions.registerListener(events -> refresh(events), TRIGGER_CREATE, TRIGGER_UPDATE, TRIGGER_REMOVE);
+            extensions.addListener(this);
+        }
+    }
+
+    private void refresh(Set<DefinitionsEvent> defEvents) {
+        boolean refresh = false;
+        for (DefinitionsEvent defEvent : defEvents) {
+            if (defEvent.getTags() != null && defEvent.getTags().get(TAG_NAME) != null
+                    && defEvent.getTags().get(TAG_NAME).equals(TAG_VALUE)) {
+                refresh = true;
+                break;
+            }
+        }
+        if (!refresh) {
+            return;
+        }
+        executor.submit(() -> {
+            try {
+                // get all of the triggers tagged for this alerter
+                Collection<Trigger> triggers = definitions.getAllTriggersByTag(TAG_NAME, TAG_VALUE);
+                log.infof("Found [%s] Tag [%s, %s] Triggers!", triggers.size(), TAG_NAME, TAG_VALUE);
+
+                Collection<FullTrigger> activeTriggers = new ArrayList<>();
+                for (Trigger trigger : triggers) {
+                    Collection<Condition> conditions = null;
+                    List<Condition> activeConditions = new ArrayList<>();
+                    try {
+                        if (trigger.isEnabled()) {
```

Comment:
I agree there is not an easy way for this. As there might be some cornercases in both directions.
But I think is better to eliminate this filter and let all triggers to be part of the extension, and deal with the resulting events in the downstream, as it is designed for it.
So, I will update this with the suggested comments.